### ISSUE TITLE
Go fileserver: Folder or File is not deletable when there is a spurious utf-8 char inside the filename

### ISSUE THREAD

**Issue by kiorky** [haiwen/seafile#2642]
reopening of #2584 & 2594

tested now with `9.0.10` but also with  `>=9.0.4`) seems not to address the issue, my already uploaded and broken files are still not deletable both with the C and GO fileservers.

consider this structure using 9.0.4 Go fileserver BEFORE upgrading to latest
```sh
mkdir folder1é toto
for i in folder1é.md toto/folder1é.md folder1é/toto.md;do echo 123 > $i;done
```

drag'drop ujpload them to your library

upgrade to 9.0.7 or latest

The first level bizarious-mis-accentuated files & folders are still not deletable, only toto folder will be.

**Comment by feiniks** [comment]
Helo, the new version of seafile only fixes the problem that the wrong file name will not be created. The wrong file name created by the old version of go fileserver has already been written to the storage, so the new version cannot delete theses files. If you want to delete this part of the file, you can delete it by deleting the parent directory.

**Comment by kiorky** [comment]
not possible if it's a top folder itself which is mis-accentuated

**Comment by feiniks** [comment]
Hello @kiorky, can you rename this folder and delete it later ?

**Comment by kiorky** [comment]
As i say, no.

**Comment by feiniks** [comment]
Hello @kiorky, if it cannot be renamed, and must delete these files, it can only be solved by migrating the current library. Files created by the previous version due to encoding problems cannot be deleted by the new version. The changes in the new version only fix the encoding problem and will not create wrongly encoded files.

**Comment by kiorky** [comment]
Yes, i know, that what even what i said on my initial report: both for that state is currently not to handle already corrupted libs and subsequently migrating them to a new, fixed, layout.

Also,i you shared the lib link with customers or any thirdparty, that's not always feasible to give them an updated library endpoint...

OTH, and finally, it's just bad to say that for library which is totally corrupted for a mishandled utf8 chars, the only solution is to manually migrate it to a non bugged one...
It should just be recoverable either by an adhoc script or an automatic migration.

**Comment by kiorky** [comment]
Why this bug is again closed, but not fixed ... !?
